apps/s_client.c

```diff
@@ -104,6 +107,12 @@ static BIO *bio_c_out = NULL;
 static int c_quiet = 0;
 static char *sess_out = NULL;
 static SSL_SESSION *psksess = NULL;
+/*
+ * TODO(QUIC): This needs to be global so we can query it easily. It would be
+ *             better if there was an SSL_is_quic() function like there is
```

Comment:
Adding SSL_is_quic() as a separate PR first would make sense to me.
If that was done perhaps an SSL_is_tls() or perhaps a generic function to return the protocol class/type as an integer.

Internally we have SSL_CONNECTION_IS_TLS13(), SSL_CONNECTION_IS_QUIC() and SSL_CONNECTION_IS_DTLS()

SSL_get_version() is "close" to this and what many applications use to report information - but that won't return anything useful in the context of QUIC and perhaps it should.

Reply:
I had already come to the same conclusion and hence #20650.

Reply:
Now that #20650 has been merged I've updated this PR to use it and removed this TODO.